apps/livestorm/src/inngest/functions/users/sync-users.ts

```diff
@@ -0,0 +1,116 @@
+import { type User } from '@elba-security/sdk';
+import { NonRetriableError } from 'inngest';
+import { eq } from 'drizzle-orm';
+import { logger } from '@elba-security/logger';
+import { type LivestormUser, getUsers } from '@/connectors/users';
+import { db } from '@/database/client';
+import { organisationsTable } from '@/database/schema';
+import { env } from '@/common/env';
+import { inngest } from '@/inngest/client';
+import { decrypt } from '@/common/crypto';
+import { createElbaClient } from '@/connectors/elba/client';
+
+const formatElbaUserDisplayName = (user: LivestormUser['attributes']) => {
+  if (user.first_name || user.last_name) {
+    return `${user.first_name || ''} ${user.last_name || ''}`.trim();
+  }
+
+  return user.email;
+};
+
+const formatElbaUser = ({ id, attributes }: LivestormUser): User => ({
+  id,
+  displayName: formatElbaUserDisplayName(attributes),
+  role: attributes.role,
+  email: attributes.email,
+  authMethod: undefined,
+  additionalEmails: [],
+});
+
+export const syncUsersPage = inngest.createFunction(
+  {
+    id: 'livestorm-sync-users',
+    priority: {
+      run: 'event.data.isFirstSync ? 600 : 0',
+    },
+    concurrency: {
+      key: 'event.data.organisationId',
+      limit: env.LIVESTORM_USERS_SYNC_CONCURRENCY,
```

Comment:
```suggestion
      limit: 1,
```
(skippable)

Because we can only a page at once, I don't see the benefit of having an other value than 1.

What could make sense is something like that:
```
concurrency: [
  { limit: env.LIVESTORM_USERS_SYNC_CONCURRENCY // like 20 },
  {
      key: 'event.data.organisationId',
      limit: env.LIVESTORM_USERS_SYNC_CONCURRENCY,
  }
]
```
But it's depends on how the ratelimit works with the Saas. I guess it's per tenant for Livestorm. so configuring a global concurrency don't really make sense.

Reply:
I agree, We  sync only one page per organisation, therefore I believe defining the concurrency is point less here,  because it will create next page once the existing one is finished , so we can ensure only one sync will be created all the time,  I don't understand how `LIVESTORM_USERS_SYNC_CONCURRENCY ` can be 20 ?